source/util.h

```diff
@@ -20,32 +20,37 @@
 #ifndef RRC_UTIL_H
 #define RRC_UTIL_H
 
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+
 #define _RRC_STRING(s) #s
 
-#define RRC_ASSERT(condition, what)                                     \
-    do                                                                  \
-    {                                                                   \
-        int st = condition;                                             \
-        char *st_str = _RRC_STRING(condition);                          \
-        if (!st)                                                        \
-        {                                                               \
-            printf("%s: assert failed: %s (got %i)", what, st_str, st); \
-            usleep(1000000);                                            \
-            exit(1);                                                    \
```

Comment:
Extracted this "print, sleep, exit" part out into its own `FATAL()` macro since there were a few places where I wanted to printf an error and exit but also use `%d` where this assert macro wasn't really usable